File: ios/Classes/PullToRefresh/PullToRefreshOptions.swift
//
//  PullToRefreshOptions.swift
//  flutter_inappwebview
//
//  Created by Lorenzo Pichilli on 03/03/21.
//

import Foundation

public class PullToRefreshOptions : Options<PullToRefreshControl> {
    
    var enabled = true
    var color: String?
    var backgroundColor: String?
    var attributedTitle: [String: Any?]?
    
    override init(){
        super.init()
    }
    
    override func parse(options: [String: Any?]) -> PullToRefreshOptions {
        let _ = super.parse(options: options)
        if let attributedTitle = options["attributedTitle"] as? [String: Any?] {
            self.attributedTitle = attributedTitle
        }
        return self
    }
    
    override func getRealOptions(obj: PullToRefreshControl?) -> [String: Any?] {
        let realOptions: [String: Any?] = toMap()
        return realOptions
    }
}

File: android/src/main/java/com/pichillilorenzo/flutter_inappwebview/pull_to_refresh/PullToRefreshOptions.java
package com.pichillilorenzo.flutter_inappwebview.pull_to_refresh;

import androidx.annotation.Nullable;

import com.pichillilorenzo.flutter_inappwebview.Options;

import java.util.HashMap;
import java.util.Map;

public class PullToRefreshOptions implements Options<PullToRefreshLayout> {
  public static final String LOG_TAG = "PullToRefreshOptions";

  public Boolean enabled = true;
  @Nullable
  public String color;
  @Nullable
  public String backgroundColor;
  @Nullable
  public Integer distanceToTriggerSync;
  @Nullable
  public Integer slingshotDistance;
  @Nullable
  public Integer size;

  public PullToRefreshOptions parse(Map<String, Object> options) {
    for (Map.Entry<String, Object> pair : options.entrySet()) {
      String key = pair.getKey();
      Object value = pair.getValue();
      if (value == null) {
        continue;
      }

      switch (key) {
        case "enabled":
          enabled = (Boolean) value;
          break;
        case "color":
          color = (String) value;
          break;
        case "backgroundColor":
          backgroundColor = (String) value;
          break;
        case "distanceToTriggerSync":
          distanceToTriggerSync = (Integer) value;
          break;
        case "slingshotDistance":
          slingshotDistance = (Integer) value;
          break;
        case "size":
          size = (Integer) value;
          break;
      }
    }

    return this;
  }

  public Map<String, Object> toMap() {
    Map<String, Object> options = new HashMap<>();
    options.put("enabled", enabled);
    options.put("color", color);
    options.put("backgroundColor", backgroundColor);
    options.put("distanceToTriggerSync", distanceToTriggerSync);
    options.put("slingshotDistance", slingshotDistance);
    options.put("size", size);
    return options;
  }

  @Override
  public Map<String, Object> getRealOptions(PullToRefreshLayout pullToRefreshLayout) {
    Map<String, Object> realOptions = toMap();
    return realOptions;
  }

}